frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Service)}>
+                {receiver.Service}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Date)}>
+                {receiver.Date}
+            </td>
+            <td
+                role="button"
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.ReportId}:`,
+                        body: receiver.ReportId,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.ReportId}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Main)}>
+                <p
+                    className={classnames(
+                        "font-mono-sm border-1px bg-primary-lighter radius-md padding-left-1 padding-right-1 margin-top-0",
+                        {
+                            "bg-blue-5": uppercaseMain === StatusEnum.BATCH,
+                            "bg-blue-10": uppercaseMain === StatusEnum.PROCESS,
+                            "bg-blue-20": uppercaseMain === StatusEnum.READY,
+                        }
+                    )}
+                >
+                    {uppercaseMain}
+                </p>
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Sub)}>
+                {receiver.Sub}
+            </td>
+            <td
+                role="button"
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.FileName
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.FileName}:`,
+                        body: receiver.FileName,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.FileName}
+            </td>
+            <td
+                role="button"
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.TransportResults
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.TransportResults}:`,
+                        body: receiver.TransportResults,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.TransportResults}
+            </td>
+        </tr>
+    );
 };
 
 export const MessageReceivers = ({ receiverDetails }: MessageReceiverProps) => {
+    const modalRef = useRef<ModalRef>(null);
+    const [modalText, setModalText] = useState({ title: "", body: "" });
```

Comment:
Oooooh already using state for an object 😆 

(Nah, I get it -- title and body should both change atomically :))

Reply:
You got me! My thinking was when it's changing a piece of how something works vs. strings. Also, as you said `title` and `body` MUST be changed at the same time, while column name and sort direction can change independent of each other. 🤷